### PR TITLE
implements list publications command

### PR DESCRIPTION
**Pull request by brunocalza**
Straightforward implementation of the `publication list` command

It goes together with the server implementation https://github.com/tablelandnetwork/basin-provider/pull/9

### CODE REVIEW

File: cmd/basin/publication.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ func newPublicationCommand() *cli.Command {
 			newPublicationCreateCommand(),
 			newPublicationStartCommand(),
 			newPublicationUploadCommand(),
+			newPublicationListCommand(),
```
Comment (brunocalza) [inline cmd/basin/publication.go:42]: adds new command

File: pkg/basinprovider/provider_test.go [hunk 2]
```diff
@@ -18,14 +18,22 @@ import (
 	"google.golang.org/grpc/test/bufconn"
 )
 
-func TestBasinProvider_Create(t *testing.T) {
+func TestBasinProvider_CreateAndList(t *testing.T) {
```
Comment (brunocalza) [inline pkg/basinprovider/provider_test.go:21]: enhances test to use `List`

File: pkg/basinprovider/provider.go [hunk 3]
```diff
@@ -203,6 +203,37 @@ func (bp *BasinProvider) Upload(
 	return nil
 }
 
+// List lists all publications from a given address.
+func (bp *BasinProvider) List(ctx context.Context, owner common.Address) ([]string, error) {
```
Comment (brunocalza) [inline pkg/basinprovider/provider.go:207]: capnp client implementation

File: README.md [hunk 4]
```diff
@@ -112,6 +113,14 @@ Then, use `basin publication upload` to upload a Parquet file.
 basin publication upload --name [namespace.relation_name] --private-key [PRIVATE_KEY] filepath
 ```
 
+## Listing Publications
+
+You can list the publications from an address by running:
+
+```bash
+basin publication list --address [ETH_ADDRESS]
```
Comment (avichalp) [inline README.md:121]: nit: there are two flags in the command, should we add an example here that uses both flags?

**Review by avichalp — APPROVED** [review]
LGTM

File: README.md (hunk 4 above)
Comment (brunocalza) [reply README.md:121]: yeah i did not included because it's optional but i can add it

**Pull request merged**